src/test/ensemble_config_change_stress.cpp

```diff
@@ -0,0 +1,668 @@
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, RemovePipelineDefinitionDuringLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_ANYMORE};         // we expect to stop creating pipelines
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK};  // we expect full continuouity of operation
+    std::set<StatusCode> allowedCreatePDResults = {};
+    std::set<StatusCode> allowedExecuteResults = {};
+    performStressTest(&StressPipelineConfigChanges::removePipelineDefinition,
+        performWholeConfigReload,
+        requiredCreatePDResults,
+        requiredExecuteResults,
+        allowedCreatePDResults,
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, ChangedPipelineConnectionName) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK};  // we expect full continuouity of operation
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK};   // we expect full continuouity of operation
+    std::set<StatusCode> allowedCreatePDResults = {};
+    std::set<StatusCode> allowedExecuteResults = {};
+    performStressTest(&StressPipelineConfigChanges::changeConnectionName,
+        performWholeConfigReload,
+        requiredCreatePDResults,
+        requiredExecuteResults,
+        allowedCreatePDResults,
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, AddedNewPipeline) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK};  // we expect full continuouity of operation
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK};   // we expect full continuouity of operation
+    std::set<StatusCode> allowedCreatePDResults = {};
+    std::set<StatusCode> allowedExecuteResults = {};
+    performStressTest(&StressPipelineConfigChanges::addNewPipeline,
+        performWholeConfigReload,
+        requiredCreatePDResults,
+        requiredExecuteResults,
+        allowedCreatePDResults,
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, RetireSpecificVersionUsed) {
+    // we declare specific version used (1) and latest model version policy with count=1
+    // then we add version 2 causing previous default to be retired
+    SetUpConfig(stressTestPipelineOneDummyConfigSpecificVersionUsed);
+    bool performWholeConfigReload = false;
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK,  // we expect full continuouity of operation
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET};             // we hit when all config changes finish to propagate
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK,
+        StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE};  // version is retired but pipeline not invalidated yet
+    std::set<StatusCode> allowedCreatePDResults = {};
```

Comment:
We never fill allowed with any status. Why do we need to pass it?

Reply:
Initially I though that we may intermittently hit some other status code, and this could be the place to put them.

To be removed probably,